### ISSUE TITLE
Passthrough drive into dsm?

### ISSUE THREAD

**Issue by alccai** [kroese/virtual-dsm#324]
I use pve to run a vm ，then install docker into the vm.

<img width="577" alt="image" src="https://github.com/kroese/virtual-dsm/assets/39586541/5deb72c2-2c8e-4871-9660-eec5ff600d9a">

the 32G drive used to the system of debian. and the 256G server used to mount into the dsm.

so in current architecturel, i cant direct use pve vitrual disk into the dsm， data.img is a virtualdisk into another virtualdisk.

question: can i mount pve vm like pve do in my phyics device https://pve.proxmox.com/wiki/Passthrough_Physical_Disk_to_Virtual_Machine_(VM)

**Comment by alccai** [comment]
updated:

edit disk args like this， and passthrough the disk, the volume showing up,  it's works,. 

<img width="956" alt="image" src="https://github.com/kroese/virtual-dsm/assets/39586541/dc22451b-451a-4d80-8bf8-3aadca8888b5">


<img width="1115" alt="image" src="https://github.com/kroese/virtual-dsm/assets/39586541/af6077e1-75ae-4b57-8363-8173436be5e4">

but i dont think the direct mount physics disk is a good idea, in my "physics" device, actually is a pve virutal disk, store on a zfs array, so the data is safe.

**Comment by alccai** [comment]
> So the problem is solved?

so, their are also need some feature in the script, like mount iscsi drive  direct to kvm and dynamic mount multi num of disks.

**Comment by alccai** [comment]
thx a lot , i will try it later.

**Comment by alccai** [comment]
but, their are also another idea， perhaps we can define the vdsm by shell script, rather use env variable. like below.

```
# their are some shell init some functions like add_disk, mount device.

#define disk part

add_diak("/storage/system.img")
add_disk("/storage/data.img")
add_disk("/dev/sdc1")

# define passthrough
add_device("/dev/nvidia0")

```

use shell script to combine qemu params mayebe a workable solutaion for custom device.

maybe i can try it by myself, if interest it.

**Comment by alccai** [comment]
yeah, for most of users, the env define is good, but in my opinion, is hard to maintenance, their are hard code all of it, so use env command to call the function like add_disk or add_device in preset, and for pro user, just mount a single file by docker-compose, like mount /storage.

```
    volumes:
      - ./vdsm.sh:/vdsm/vdsm.sh
```

the vdsm.sh also store outside the docker, like /storage volume, it will not take effect for current user,  but keep the api stablize and open some "inner" function to pro user, will give more freedom to use it.

**Comment by ad-on-is** [comment]
> But first, can you confirm this works in your case:
> 
> ```yaml
>         environment:
>             DEVICE: "/dev/vdc1"
>         devices:
>             - /dev/vdc1
> ```

not OP, but this works wonderfully... Thank you! Any chance we can pass through multiple-devices?

**Comment by ad-on-is** [comment]
cool... thx!

Sorry, to hijack... but do yo know of any way how to setup this up behind a reverse-proxy... 

I have caddy successfully configured with nas.example.com -> 192.168.0.100, but it only shows a blue screen saying "DSM is located at 192.168.1.40" and redirects me to the IP instead.

**Comment by ad-on-is** [comment]
Oh man... thank you so much for this one... I totally didn't see that you mentioned this.

**Comment by ad-on-is** [comment]
You are my hero! it works!
